### PR TITLE
Reorganize tests.

### PR DESCRIPTION
**Pull request by MartinEden**
This is a preliminary refactor for https://vimc.myjetbrains.com/youtrack/issue/VIMC-1054. This needs to be merged in before reviewing #119.

### CODE REVIEW

File: src/databaseTests/src/test/kotlin/org/vaccineimpact/api/databaseTests/RepositoryTestSuite.kt [hunk 1]
```diff
@@ -15,7 +17,7 @@ import org.vaccineimpact.api.test_helpers.DatabaseCreationHelper
 // Keep these sorted alphabetically, for consistency
 @RunWith(Suite::class)
 @Suite.SuiteClasses(
-        BurdenEstimateRepositoryTests::class,
+        UploadBurdenEstimateTests::class,
         BurdenEstimateRepositoryTestsDeprecated::class,
```
Comment (littledice) [inline src/databaseTests/src/test/kotlin/org/vaccineimpact/api/databaseTests/RepositoryTestSuite.kt]: You'll want to add Retrieve... tests in here too

Comment (MartinEden) [reply src/databaseTests/src/test/kotlin/org/vaccineimpact/api/databaseTests/RepositoryTestSuite.kt]: Fixed - good catch :)

**Comment by MartinEden** [comment]
I think you're probably looking at an older branch in TeamCity. The old branch (that I ended up not using) is called `i1054_reorganize_tests`. This PR is for `i1054_refactor`.

**Comment by littledice** [comment]
Ah yes, that makes more sense!

**Pull request merged**